test/ex_doc/formatter/html_test.exs

```diff
@@ -112,6 +112,24 @@ defmodule ExDoc.Formatter.HTMLTest do
     assert LazyHTML.text(bar_content["h1"]) == "README bar"
   end
 
+  test "extras defined as external urls", %{tmp_dir: tmp_dir} = context do
+    config =
```

Comment:
I think we should add a test that shows grouping work. Imagine you want all URLs in a section in the sidebar called Important Links. The `Template.sidebar_extras` kinda implies URLs work with groups but we need to be sure it works "end to end".

Reply:
It seems groups expect the filename but there is none here, so we probably have to pass the filename or the url (and update its docs accordingly).

Reply:
Grouping does work based on the input name, but it doesn't consider the URL. For example, with the extra `[Elixir: [url: "https://elixir-lang.org"]]` this will match on "Elixir", but it doesn't consider the url value at all.

```elixir
    input = to_string(input)
    title = input_options[:title] || input
    group = GroupMatcher.match_extra(groups, input)
```

That's a little different than how regular grouping works because the input lacks path information for easy grouping.

I've opted to expand `GroupMatcher.match_extra` to be url/path aware, so the pattern applies to either the title or url.